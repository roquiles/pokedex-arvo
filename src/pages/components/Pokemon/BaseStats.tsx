type BaseStatsProps = {
  stats: Array<{
    name: string;
    value: number;
  }>;
};

const BaseStats: React.FC<BaseStatsProps> = ({ stats }) => {
  const statsValues = stats?.map((stat) => stat.value);
  const maxStatValue = Math.max(...statsValues);

  const getStatWidth = (value: number) =>
    value === maxStatValue
      ? '100%'
      : `${Math.round((value * 100) / maxStatValue)}%`;

  return (
    <div className="col-span-3 md:col-span-2 text-start ml-4 sm:ml-8">
      <h3 className=" text-2xl text-gray-800 font-semibold mb-4 ">
        Base Stats
      </h3>
      {stats?.map(({ name, value }) => (
        <div key={name} className="mb-2">
          <span className="uppercase text-gray-500">{name}</span>
          <div className="flex">
            <div
              className="h6 rounded-sm bg-teal-600 mr-1"
              style={{
                width: getStatWidth(value),
              }}
            />
            <span>{value}</span>
          </div>
        </div>
      ))}
    </div>
  );
};

export default BaseStats;
