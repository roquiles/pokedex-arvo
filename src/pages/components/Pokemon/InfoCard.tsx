import { PokemonType } from '@/types/pokemon';

const InfoCard: React.FC<{ infos: Partial<PokemonType> }> = ({ infos }) => {
  const { weight, height, abilities } = infos;

  return (
    <div className="rounded-lg bg-yellow-400 p-5 text-gray-600 mb-8">
      <div className="flex flex-col mb-3.5 items-center">
        Height:&nbsp;
        <span className="text-gray-800 text-xl font-semibold">{height}</span>
      </div>
      <div className="flex flex-col mb-3.5 items-center">
        Weight:&nbsp;
        <span className="text-gray-800 text-xl font-semibold">{weight}</span>
      </div>
      <div className="flex flex-col mb-3.5 text-center text-wrap">
        Abilities:&nbsp;
        <ul>
          {abilities?.map((ability) => (
            <li
              className="capitalize text-gray-800 text-xl mr-1 font-semibold"
              key={ability}
            >
              {ability}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default InfoCard;
