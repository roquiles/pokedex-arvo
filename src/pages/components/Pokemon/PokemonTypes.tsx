import { getTypeColor } from '@/utils';

const PokemonTypes: React.FC<{ types: string[] }> = ({ types }) => (
  <div className="ml-4 text-start">
    <h3 className="text-2xl text-gray-800 font-semibold mb-4">Types</h3>
    {types.map((item) => (
      <span
        style={{ backgroundColor: getTypeColor(item) }}
        className="px-6 py-1 rounded-full mr-2"
        key={item}
      >
        {item}
      </span>
    ))}
  </div>
);

export default PokemonTypes;
