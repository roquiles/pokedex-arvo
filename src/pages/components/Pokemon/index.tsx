import { PokemonType } from '@/types/pokemon';
import Image from 'next/image';
import InfoCard from './InfoCard';
import PokemonTypes from './PokemonTypes';
import BaseStats from './BaseStats';

type PokemonProps = {
  activePokemon: PokemonType;
};

const Pokemon: React.FC<PokemonProps> = ({ activePokemon }) => {
  const { name, id, height, weight, abilities, types, stats } = activePokemon;

  return (
    <div className="bg-white rounded mt-10 p-6">
      <div className="flex items-center flex-col mb-5 sm:flex-row sm:ml-6 ">
        <Image
          width={150}
          height={150}
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`}
          alt={name}
        />
        <h2 className="text-gray-700 capitalize text-4xl font-semibold sm:ml-5">
          #{id} - {name}
        </h2>
      </div>

      <div className="w-full grid grid-cols-3">
        <div className="col-span-3 mt-10 md:col-span-1 md:mt-0">
          <InfoCard infos={{ height, weight, abilities }} />
          <PokemonTypes types={types} />
        </div>
        <BaseStats stats={stats} />
      </div>
    </div>
  );
};

export default Pokemon;
