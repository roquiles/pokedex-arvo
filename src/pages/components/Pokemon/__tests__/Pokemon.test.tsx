import { render } from '@testing-library/react';
import Pokemon from '..';

const pokemonMock = {
  id: 113,
  name: 'Chansey',
  height: 11,
  weight: 346,
  abilities: ['natural-cure', 'serene-grace', 'healer'],
  types: ['normal'],
  stats: [
    {
      name: 'hp',
      value: 250,
    },
    { name: 'attack', value: 5 },
    { name: 'defense', value: 5 },
    { name: 'special-attack', value: 35 },
    { name: 'special-defense', value: 105 },
    { name: 'speed', value: 50 },
  ],
};

describe('', () => {
  it('should render correctly', () => {
    const component = render(<Pokemon activePokemon={pokemonMock} />);
    expect(component).toMatchSnapshot();
  });
});
