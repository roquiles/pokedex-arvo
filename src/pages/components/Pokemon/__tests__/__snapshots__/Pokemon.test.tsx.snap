// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white rounded mt-10 p-6"
      >
        <div
          class="flex items-center flex-col mb-5 sm:flex-row sm:ml-6 "
        >
          <img
            alt="Chansey"
            height="150"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/113.png"
            width="150"
          />
          <h2
            class="text-gray-700 capitalize text-4xl font-semibold sm:ml-5"
          >
            #
            113
             - 
            Chansey
          </h2>
        </div>
        <div
          class="w-full grid grid-cols-3"
        >
          <div
            class="col-span-3 mt-10 md:col-span-1 md:mt-0"
          >
            <div
              class="rounded-lg bg-yellow-400 p-5 text-gray-600 mb-8"
            >
              <div
                class="flex flex-col mb-3.5 items-center"
              >
                Height: 
                <span
                  class="text-gray-800 text-xl font-semibold"
                >
                  11
                </span>
              </div>
              <div
                class="flex flex-col mb-3.5 items-center"
              >
                Weight: 
                <span
                  class="text-gray-800 text-xl font-semibold"
                >
                  346
                </span>
              </div>
              <div
                class="flex flex-col mb-3.5 text-center text-wrap"
              >
                Abilities: 
                <ul>
                  <li
                    class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                  >
                    natural-cure
                  </li>
                  <li
                    class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                  >
                    serene-grace
                  </li>
                  <li
                    class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                  >
                    healer
                  </li>
                </ul>
              </div>
            </div>
            <div
              class="mb-10 text-start sm:ml-4"
            >
              <h3
                class="text-2xl text-gray-800 font-semibold mb-4"
              >
                Types
              </h3>
              <span
                class="px-6 py-1 rounded-full mr-2"
                style="background-color: rgb(217, 217, 217);"
              >
                normal
              </span>
            </div>
          </div>
          <div
            class="col-span-3 md:col-span-2 text-start sm:ml-8"
          >
            <h3
              class=" text-2xl text-gray-800 font-semibold mb-4 "
            >
              Base Stats
            </h3>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                hp
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 100%;"
                />
                <span>
                  250
                </span>
              </div>
            </div>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                attack
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 2%;"
                />
                <span>
                  5
                </span>
              </div>
            </div>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                defense
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 2%;"
                />
                <span>
                  5
                </span>
              </div>
            </div>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                special-attack
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 14%;"
                />
                <span>
                  35
                </span>
              </div>
            </div>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                special-defense
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 42%;"
                />
                <span>
                  105
                </span>
              </div>
            </div>
            <div
              class="mb-2"
            >
              <span
                class="uppercase text-gray-500"
              >
                speed
              </span>
              <div
                class="flex"
              >
                <div
                  class="h6 rounded-sm bg-teal-600 mr-1"
                  style="width: 20%;"
                />
                <span>
                  50
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white rounded mt-10 p-6"
    >
      <div
        class="flex items-center flex-col mb-5 sm:flex-row sm:ml-6 "
      >
        <img
          alt="Chansey"
          height="150"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/113.png"
          width="150"
        />
        <h2
          class="text-gray-700 capitalize text-4xl font-semibold sm:ml-5"
        >
          #
          113
           - 
          Chansey
        </h2>
      </div>
      <div
        class="w-full grid grid-cols-3"
      >
        <div
          class="col-span-3 mt-10 md:col-span-1 md:mt-0"
        >
          <div
            class="rounded-lg bg-yellow-400 p-5 text-gray-600 mb-8"
          >
            <div
              class="flex flex-col mb-3.5 items-center"
            >
              Height: 
              <span
                class="text-gray-800 text-xl font-semibold"
              >
                11
              </span>
            </div>
            <div
              class="flex flex-col mb-3.5 items-center"
            >
              Weight: 
              <span
                class="text-gray-800 text-xl font-semibold"
              >
                346
              </span>
            </div>
            <div
              class="flex flex-col mb-3.5 text-center text-wrap"
            >
              Abilities: 
              <ul>
                <li
                  class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                >
                  natural-cure
                </li>
                <li
                  class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                >
                  serene-grace
                </li>
                <li
                  class="capitalize text-gray-800 text-xl mr-1 font-semibold"
                >
                  healer
                </li>
              </ul>
            </div>
          </div>
          <div
            class="mb-10 text-start sm:ml-4"
          >
            <h3
              class="text-2xl text-gray-800 font-semibold mb-4"
            >
              Types
            </h3>
            <span
              class="px-6 py-1 rounded-full mr-2"
              style="background-color: rgb(217, 217, 217);"
            >
              normal
            </span>
          </div>
        </div>
        <div
          class="col-span-3 md:col-span-2 text-start sm:ml-8"
        >
          <h3
            class=" text-2xl text-gray-800 font-semibold mb-4 "
          >
            Base Stats
          </h3>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              hp
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 100%;"
              />
              <span>
                250
              </span>
            </div>
          </div>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              attack
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 2%;"
              />
              <span>
                5
              </span>
            </div>
          </div>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              defense
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 2%;"
              />
              <span>
                5
              </span>
            </div>
          </div>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              special-attack
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 14%;"
              />
              <span>
                35
              </span>
            </div>
          </div>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              special-defense
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 42%;"
              />
              <span>
                105
              </span>
            </div>
          </div>
          <div
            class="mb-2"
          >
            <span
              class="uppercase text-gray-500"
            >
              speed
            </span>
            <div
              class="flex"
            >
              <div
                class="h6 rounded-sm bg-teal-600 mr-1"
                style="width: 20%;"
              />
              <span>
                50
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
